libsolidity/analysis/ControlFlowBuilder.cpp

```diff
@@ -318,24 +320,10 @@ bool ControlFlowBuilder::visit(ModifierInvocation const& _modifierInvocation)
 		for (auto& argument: *arguments)
 			appendControlFlow(*argument);
 
-	auto modifierDefinition = dynamic_cast<ModifierDefinition const*>(
-		_modifierInvocation.name().annotation().referencedDeclaration
-	);
-	if (!modifierDefinition) return false;
-	if (!modifierDefinition->isImplemented()) return false;
-	solAssert(!!m_returnNode, "");
-
-	m_placeholderEntry = newLabel();
-	m_placeholderExit = newLabel();
-
-	appendControlFlow(*modifierDefinition);
-	connect(m_currentNode, m_returnNode);
+	modifierInvocations.insert(&_modifierInvocation);
 
-	m_currentNode = m_placeholderEntry;
-	m_returnNode = m_placeholderExit;
-
-	m_placeholderEntry = nullptr;
-	m_placeholderExit = nullptr;
+	m_currentNode->modifierInvocation = &_modifierInvocation;
```

Comment:
Previously, modifiers were analyzed and the control flow became part of the function, but inheritance was ignored.
Now we just remember the invocation and create for each their own flow.